### PR TITLE
Kgp 516 limit order UI improvements

### PR DESCRIPTION
**Pull request by nguyenhoaidanh**
- update new UI limit order form
- change text

### CODE REVIEW

File: src/components/CurrencyInputPanel/index.tsx [hunk 1]
```diff
@@ -129,11 +127,27 @@ const StyledBalanceMax = styled.button`
   }
 `
 
-const Card2 = styled(Card)<{ balancePosition: string }>`
+const Card2 = styled(Card)`
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx]: Better name?

File: src/components/CurrencyInputPanel/index.tsx [hunk 2]
```diff
@@ -129,11 +127,27 @@ const StyledBalanceMax = styled.button`
...
   padding: 0 0.25rem 0.5rem;
-  text-align: ${({ balancePosition }) => `${balancePosition}`};
+  text-align: right;
 `
 
+const PoolLockContent = (
+  <FixedContainer>
+    <Flex padding={'0 20px'} sx={{ gap: '16px' }}>
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:137]: Can use double quote.

File: src/components/CurrencyInputPanel/index.tsx [hunk 3]
```diff
@@ -129,11 +127,27 @@ const StyledBalanceMax = styled.button`
...
+      <div style={{ width: '26px', margin: 'auto' }}>
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:138]: ...

File: src/components/CurrencyInputPanel/index.tsx [hunk 4]
```diff
@@ -129,11 +127,27 @@ const StyledBalanceMax = styled.button`
...
+        <Lock />
+      </div>
+      <Text fontSize="12px" textAlign="left" padding="8px 16px" lineHeight={'16px'}>
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:141]: Double quote.

File: src/components/CurrencyInputPanel/index.tsx [hunk 5]
```diff
@@ -129,11 +127,27 @@ const StyledBalanceMax = styled.button`
...
+        <Trans>
+          The price of the pool is outside your selected price range and hence you can only deposit a single token. To
+          see more options, update the price range.
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:144]: Why hit new line here?

File: src/components/CurrencyInputPanel/index.tsx [hunk 6]
```diff
@@ -166,33 +179,35 @@ interface CurrencyInputPanelProps {
   error?: boolean
   maxLength?: number
   filterWrap?: boolean
+  lockIcon?: boolean
+}
+
+const noop = (value: string) => {
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:185]: I see many `noop` like this in source code, can you just modify component `NumericalInput` to accept undefined value.

File: src/components/CurrencyInputPanel/index.tsx [hunk 7]
```diff
@@ -229,37 +245,23 @@ export default function CurrencyInputPanel({
...
-      {label && (
-        <Card2 borderRadius={'20px'} balancePosition={balancePosition}>
-          <Flex justifyContent={label ? 'space-between' : 'end'} alignItems="center">
-            {label && (
-              <Text fontSize={12} color={theme.subText} fontWeight={500}>
-                {label}:
-              </Text>
-            )}
+      {label && positionLabel === 'out' && (
+        <Card2 borderRadius={'20px'}>
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx]: Double.

File: src/components/CurrencyInputPanel/index.tsx [hunk 8]
```diff
@@ -229,37 +245,23 @@ export default function CurrencyInputPanel({
...
+          <Flex justifyContent={'space-between'} alignItems="center">
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:250]: Double.

File: src/components/CurrencyInputPanel/index.tsx [hunk 9]
```diff
@@ -329,27 +332,17 @@ export default function CurrencyInputPanel({
...
-                        active={Boolean(currency && currency.symbol)}
-                        fontSize={fontSize}
-                        style={{ paddingRight: disableCurrencySelect ? '8px' : 0 }}
-                      >
-                        {(nativeCurrency?.symbol && maxCurrencySymbolLength
-                          ? shortString(nativeCurrency.symbol, maxCurrencySymbolLength)
-                          : nativeCurrency?.symbol) || <Trans>Select a token</Trans>}
-                      </StyledTokenName>
-                    )}
+                    {currency && !hideLogo ? <CurrencyLogo currency={currency} size={'20px'} /> : null}
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:335]: Double.

File: src/components/CurrencyInputPanel/index.tsx [hunk 10]
```diff
@@ -329,27 +332,17 @@ export default function CurrencyInputPanel({
...
+                    <StyledTokenName
+                      className="token-symbol-container"
```
Comment (nguyenhuudungz) [inline src/components/CurrencyInputPanel/index.tsx:337]: What's this classname for?

File: src/components/swapv2/LimitOrder/helpers.ts [hunk 11]
```diff
@@ -121,3 +123,29 @@ export const calcPercentFilledOrder = (value: string, total: string, decimals: n
...
+  const { currencyIn, currencyOut, chainId, account, inputAmount, outputAmount, expiredAt } = params
+  const parseInputAmount = tryParseAmount(inputAmount, currencyIn ?? undefined)
+  return {
+    chainId: chainId.toString(),
+    makerAsset: currencyIn?.wrapped.address,
+    takerAsset: currencyOut?.wrapped.address,
+    maker: account,
+    makingAmount: parseInputAmount?.quotient?.toString(),
+    takingAmount: tryParseAmount(outputAmount, currencyOut)?.quotient?.toString(),
+    expiredAt: (expiredAt / 1000) | 0,
```
Comment (namgold) [inline src/components/swapv2/LimitOrder/helpers.ts:149]: `| 0` mistake?
```suggestion
    expiredAt: (expiredAt / 1000) || 0,
```

File: src/components/swapv2/LimitOrder/ListOrder/SummaryNotify.tsx [hunk 12]
```diff
@@ -40,7 +40,7 @@ export default function SummaryNotify({
         {formatAmountOrder(takingAmount, takerAssetDecimals)} {takerAssetSymbol}
       </Text>{' '}
       <Text as="span" color={theme.subText}>
-        at a {takerAssetSymbol} price of {rate} {makerAssetSymbol}
+        at {takerAssetSymbol} price of {rate} {makerAssetSymbol}
```
Comment (viet-nv) [inline src/components/swapv2/LimitOrder/ListOrder/SummaryNotify.tsx:43]: should wrap with  <Trans>

File: src/components/swapv2/LimitOrder/helpers.ts (hunk 11 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/helpers.ts:149]: shortcut for Math.floor =))

File: src/components/swapv2/LimitOrder/helpers.ts [hunk 13]
```diff
@@ -121,3 +123,29 @@ export const calcPercentFilledOrder = (value: string, total: string, decimals: n
     return '0'
   }
 }
+
+export const getErrorMessage = (error: any) => {
+  console.error(error)
```
Comment (namgold) [inline src/components/swapv2/LimitOrder/helpers.ts:128]: add some context
```suggestion
  console.error('Limit order error or something', {error})
```

Comment (nguyenhuudungz) [inline src/components/swapv2/LimitOrder/helpers.ts:128]: Remove this?

**Review by nguyenhuudungz — CHANGES_REQUESTED** [review]

File: src/components/CurrencyInputPanel/index.tsx (hunk 5 above)
Comment (namgold) [reply src/components/CurrencyInputPanel/index.tsx:144]: it's automatically by lint I suppose

File: src/components/CurrencyInputPanel/index.tsx (hunk 10 above)
Comment (nguyenhoaidanh) [reply src/components/CurrencyInputPanel/index.tsx:337]: I don't know, old code.
I think for QA automation test

File: src/components/swapv2/LimitOrder/ListOrder/SummaryNotify.tsx (hunk 12 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/ListOrder/SummaryNotify.tsx:43]: Trans wrapper
image

File: src/components/swapv2/LimitOrder/helpers.ts (hunk 13 above)
Comment (nguyenhoaidanh) [reply src/components/swapv2/LimitOrder/helpers.ts]: no need

**Pull request merged**